backend/src/v5/processors/teamspaces/teamspaces.js

```diff
@@ -15,13 +15,31 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-const { getAccessibleTeamspaces } = require('../../models/users');
-const { getJobsToUsers } = require('../../models/jobs');
-const { getMembersInfo } = require('../../models/teamspaces');
+const { addDefaultJobs, assignUserToJob, getJobsToUsers } = require('../../models/jobs');
+const { createTeamspaceRole, grantTeamspaceRoleToUser } = require('../../models/roles');
+const { createTeamspaceSettings, getMembersInfo } = require('../../models/teamspaces');
+const { getAccessibleTeamspaces, grantTeamspacePermissionsToUser } = require('../../models/users');
+const { DEFAULT_JOBS } = require('../../models/jobs.constants');
 const { isTeamspaceAdmin } = require('../../utils/permissions/permissions');
+const { logger } = require('../../utils/logger');
 
 const Teamspaces = {};
 
+Teamspaces.initTeamspace = async (username) => {
+	try {
+		await createTeamspaceRole(username);
+		await Promise.all([
+			grantTeamspaceRoleToUser(username, username),
+			addDefaultJobs(username),
+			createTeamspaceSettings(username),
+			grantTeamspacePermissionsToUser(username, username, ['teamspace_admin']),
```

Comment:
`teamspace_admin` is a constant in utils/permissions.constants

avoid hard coding